File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.2/JMBA_InComm_iOS_SDK/InCommSDK/Models/Request/InCommOrderItem.swift
//
//  InCommOrderItem.swift
//  InCommSDK
//
//  Created by Taha Samad on 8/17/15.
//  Copyright (c) 2015 Fishbowl. All rights reserved.
//

import Foundation
import SwiftyJSON

public struct InCommOrderItem {
    public var amount: Double
    public var brandId: String
    public var quantity: UInt32
    //
    public var embossText: String?
    public var embossTextId: String?
    public var expirationDate: Date?
    public var imageCode: String?
    public var messageFrom: String?
    public var messageText: String?
    public var messageTo: String?
    public var productId: Int32?
    public var productName: String?
    public var hideAmount:Bool?

    public init(brandId: String, amount: Double, quantity: UInt32) {
        self.amount   = amount
        self.brandId  = brandId
        self.quantity = quantity
    }
    
    public func serializeAsJSONDictionary() -> InCommJSONDictionary {
        var jsonDict               = InCommJSONDictionary()
        jsonDict["Amount"]         = amount as AnyObject?
        jsonDict["BrandId"]        = brandId as AnyObject?
        jsonDict["Quantity"]       = NSNumber(value: quantity as UInt32)
        //
        jsonDict["EmbossText"]     = embossText as AnyObject?
        jsonDict["EmbossTextId"]   = embossTextId as AnyObject?
        jsonDict["ExpirationDate"] = expirationDate?.InCommDateTimeFormatString() as AnyObject?
        jsonDict["ImageCode"]      = imageCode as AnyObject?
        jsonDict["MessageFrom"]    = messageFrom as AnyObject?
        jsonDict["MessageText"]    = messageText as AnyObject?
        jsonDict["MessageTo"]      = messageTo as AnyObject?
        jsonDict["ProductId"]      = productId != nil ? NSNumber(value: productId! as Int32) : nil
        jsonDict["ProductName"]    = productName as AnyObject?
        jsonDict["HideAmount"]     = hideAmount as AnyObject?
        return jsonDict
    }
    
}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.2/JMBA_InComm_Android_SDK/app/src/main/java/com/wearehathway/apps/incomm/Models/InCommOrderItem.java
package com.wearehathway.apps.incomm.Models;

import java.util.Date;

/**
 * Created by Nauman Afzaal on 13/08/15.
 */
public class InCommOrderItem
{
    private String Id;
    private double Amount;
    private String BrandId;
    private String EmbossText;
    private int EmbossTextId;
    private Date ExpirationDate;
    private String ImageCode;
    private String MessageFrom;
    private String MessageText;
    private String MessageTo;
    private int ProductId;
    private String ProductName;
    private int Quantity;
    private Boolean HideAmount;

    public double getAmount()
    {
        return Amount;
    }

    public void setAmount(double amount)
    {
        Amount = amount;
    }

    public String getBrandId()
    {
        return BrandId;
    }

    public void setBrandId(String brandId)
    {
        BrandId = brandId;
    }

    public String getEmbossText()
    {
        return EmbossText;
    }

    public void setEmbossText(String embossText)
    {
        EmbossText = embossText;
    }

    public int getEmbossTextId()
    {
        return EmbossTextId;
    }

    public void setEmbossTextId(int embossTextId)
    {
        EmbossTextId = embossTextId;
    }

    public Date getExpirationDate()
    {
        return ExpirationDate;
    }

    public void setExpirationDate(Date expirationDate)
    {
        ExpirationDate = expirationDate;
    }

    public String getImageCode()
    {
        return ImageCode;
    }

    public void setImageCode(String imageCode)
    {
        ImageCode = imageCode;
    }

    public String getMessageFrom()
    {
        return MessageFrom;
    }

    public void setMessageFrom(String messageFrom)
    {
        MessageFrom = messageFrom;
    }

    public String getMessageText()
    {
        return MessageText;
    }

    public void setMessageText(String messageText)
    {
        MessageText = messageText;
    }

    public String getMessageTo()
    {
        return MessageTo;
    }

    public void setMessageTo(String messageTo)
    {
        MessageTo = messageTo;
    }

    public int getProductId()
    {
        return ProductId;
    }

    public void setProductId(int productId)
    {
        ProductId = productId;
    }

    public String getProductName()
    {
        return ProductName;
    }

    public void setProductName(String productName)
    {
        ProductName = productName;
    }

    public int getQuantity()
    {
        return Quantity;
    }

    public void setQuantity(int quantity)
    {
        Quantity = quantity;
    }

    public String getId()
    {
        return Id;
    }

    public void setId(String id)
    {
        Id = id;
    }

    public Boolean getHideAmount() {
        return HideAmount;
    }

    public void setHideAmount(Boolean hideAmount) {
        HideAmount = hideAmount;
    }
}
